@rajkan01 I can't add you as a reviewer, but since I'm touching the file you've modified lately, I'm asking for a review here. I'm talking specifically about this commit:
https://github.com/zephyrproject-rtos/zephyr/pull/60789/commits/ce8689efc8c60472bdb3e4dcd19970baf91b3344
Building that file with TFM enabled caused the linker to throw redefinition erros, as the functions the file provides are also defined in TFM crypto
> home/robert/repos/zephyrproject/zephyr/samples/hello_world/build/tfm/install/interface/src/tfm_crypto_api.c:32: multiple definition of `psa_open_key'; modules/mbedtls/libmbedTLSCrypto.a(psa_crypto_slot_management.c.obj):/home/robert/repos/zephyrproject/modules/crypto/mbedtls/library/psa_crypto_slot_management.c:498: first defined here

I've checked with @joerchan and according to him, other files should also be conditionally excluded when TFM is enabled, similarily to what was done in NCS `nrf_security` module:
https://github.com/nrfconnect/sdk-nrfxlib/blob/v2.4.0/nrf_security/src/core/builtin/CMakeLists.txt#L37C19-L37C19
Other files did not cause any trouble in this particular case but this likely is something to the TFM team to investigate.

--- a/modules/mbedtls/Kconfig.tls-generic
+++ b/modules/mbedtls/Kconfig.tls-generic
@@ -422,6 +422,7 @@ config MBEDTLS_SERVER_NAME_INDICATION
 
 config MBEDTLS_PK_WRITE_C
 	bool "The generic public (asymmetric) key writer"
+	default y if MBEDTLS_PSA_CRYPTO_C
 	help
 	  Enable generic public key write functions.
 

--- a/modules/mbedtls/configs/config-tls-generic.h
+++ b/modules/mbedtls/configs/config-tls-generic.h
@@ -394,15 +394,6 @@
 #define MBEDTLS_X509_USE_C
 #endif
 
-#if defined(MBEDTLS_X509_USE_C) || \
-    defined(MBEDTLS_ECDSA_C)
-#define MBEDTLS_ASN1_PARSE_C
-#endif
-
-#if defined(MBEDTLS_ECDSA_C)
-#define MBEDTLS_ASN1_WRITE_C
-#endif
-
 #if defined(MBEDTLS_DHM_C) || \
     defined(MBEDTLS_ECP_C) || \
     defined(MBEDTLS_RSA_C) || \
@@ -426,6 +417,14 @@
 
 #if defined(MBEDTLS_PK_PARSE_C) || defined(MBEDTLS_PK_WRITE_C)
 #define MBEDTLS_PK_C
+#endif
+
+#if defined(MBEDTLS_X509_USE_C) || defined(MBEDTLS_ECDSA_C)
+#define MBEDTLS_ASN1_PARSE_C
+#endif
+
+#if defined(MBEDTLS_ECDSA_C) || defined(MBEDTLS_PK_WRITE_C)
+#define MBEDTLS_ASN1_WRITE_C
 #endif
 
 #if defined(CONFIG_MBEDTLS_PKCS5_C)

--- a/modules/uoscore-uedhoc/CMakeLists.txt
+++ b/modules/uoscore-uedhoc/CMakeLists.txt
@@ -27,6 +27,12 @@ if (CONFIG_UOSCORE OR CONFIG_UEDHOC)
 
   zephyr_library_link_libraries(mbedTLS)
 
+  if (CONFIG_BUILD_WITH_TFM)
+    zephyr_library_include_directories(
+      $<TARGET_PROPERTY:tfm,TFM_BINARY_DIR>/install/interface/include
+    )
+  endif()
+
 # UOSCORE
 
   if (CONFIG_UOSCORE)

--- a/tests/lib/uoscore/testcase.yaml
+++ b/tests/lib/uoscore/testcase.yaml
@@ -1,0 +1,6 @@
+tests:
+  libraries.uoscore:
+    min_flash: 240
+    tags: net crypto uoscore
+    integration_platforms:
+      - qemu_x86

